libr/bin/bfile.c

```diff
@@ -904,3 +906,24 @@ R_API int r_bin_file_hash(RBin *bin, ut64 limit, const char *file) {
 	free (buf);
 	return true;
 }
+
+R_API RBinFile *r_bin_file_at(RBin *bin, ut64 addr) {
+	RBinFile *bf;
+	RListIter *iter;
+	RBinFile *obf = bin->cur; // XXX this must die
+	r_list_foreach (bin->binfiles, iter, bf) {
+		bin->cur = bf;
+		RList *sections = r_bin_get_sections (bin);
+		RBinSection *s;
+		r_list_foreach (sections, iter, s) {
+			ut64 from = s->vaddr;
+			ut64 to = from + s->vsize;
+			if (R_BETWEEN (from, addr, to)) {
```

Comment:
the bad about R_BETWEEN is that it includes both the start and the end, so i think you should use `to - 1`